### PR TITLE
Don't allow standalone KRA uninstalls

### PR DESCRIPTION
**Pull request by stlaz**
KRA uninstallation is very likely to break the user's setup. Don't
allow it at least till we can be safely sure we are able to remove
it in a standalone manner without breaking anything.

https://pagure.io/freeipa/issue/6538

### CODE REVIEW

File: ipaserver/install/ipa_kra_install.py [hunk 1]
```diff
@@ -83,33 +83,13 @@ def validate_options(self, needs_root=True):
     @classmethod
     def get_command_class(cls, options, args):
         if options.uninstall:
-            return KRAUninstaller
+            raise RuntimeError(
```
Comment (tomaskrizek) [inline ipaserver/install/ipa_kra_install.py]: `ipa-kra-install --uninstall` prints a traceback instead of just showing the error message.

**Comment by tomaskrizek** [comment]
You should also remove:

```
ipaplatform/base/paths.py:313:    IPASERVER_KRA_UNINSTALL_LOG
ipatests/test_integration/tasks.py:71:    host.collect_log(paths.IPASERVER_KRA_UNINSTALL_LOG)
ipatests/test_integration/tasks.py:73:    host.collect_log(paths.IPASERVER_KRA_UNINSTALL_LOG)
ipatests/test_integration/test_vault.py:145:    def test_create_and_retrieve_vault_after_kra_uninstall_on_replica
```

File: ipaserver/install/ipa_kra_install.py [hunk 2]
```diff
@@ -83,33 +83,13 @@ def validate_options(self, needs_root=True):
...
+                'Standalone KRA uninstalling was removed in FreeIPA 4.5 as it '
+                'had never worker properly and only caused issues.')
```
Comment (stlaz) [inline ipaserver/install/ipa_kra_install.py]: Perhaps `uninstallation` is the better word here, thanks for finding the typo as well. However, `had` is used properly, check your grammar ;)

Comment (tomaskrizek) [inline ipaserver/install/ipa_kra_install.py]: Grammar has been checked. You're indeed correct :)

**Comment by stlaz** [comment]
Should be fixed now, had to add `sys.exit()` call not to show traceback 😞

**Comment by MartinBasti** [comment]
`ScriptError` didn't work?

**Comment by stlaz** [comment]
@MartinBasti unfortunately not.

**Comment by MartinBasti** [comment]
Waiting for more opinions about removing KRA --uninstall

**Comment by pvoborni** [comment]
OK, so this pr remove `--uninstall` from `ipa-kra-install`.  Did it work in the past? Or it always broke the installation? AFAIK this workflow was not really tested. If answers are "No, Yes, Yes" then I'm OK with the PR.

File: ipaserver/install/ipa_kra_install.py [hunk 3]
```diff
@@ -83,33 +84,14 @@ def validate_options(self, needs_root=True):
     @classmethod
     def get_command_class(cls, options, args):
         if options.uninstall:
-            return KRAUninstaller
+            sys.exit(
+                'ERROR: Standalone KRA uninstallation was removed in '
+                'FreeIPA 4.5 as it had never worked properly and only caused '
```
Comment (MartinBasti) [inline ipaserver/install/ipa_kra_install.py]: I would change the wording to `ERROR: Standalone KRA uninstallation was removed in FreeIPA 4.5` without extra information about how broken it was

Comment (stlaz) [inline ipaserver/install/ipa_kra_install.py]: Some kind of reason should be provided. Perhaps we can reword it to "...as it was a potential cause of issues."

**Comment by MartinBasti** [comment]
needs rebase

**Comment by stlaz** [comment]
Rebased.

**Comment by MartinBasti** [comment]
master:

* 5d3a0e6758866239c886e998a6d89c5a4b150184 Don't allow standalone KRA uninstalls
